### PR TITLE
NA-MD in the case where electronic back reaction is not included can be executed with low computational cost.

### PR DESCRIPTION
**Pull request by kosukesato**
# Major change
So far, the program has been coded to execute NA-MD in the case where electronic back reaction is included. 
 However, as a defect, it is not proper when executing NA-MD in another case: the back reaction is not included. In this case, we can't reduce the number of SCF calculation which needs the largest amount of time. For reducing the computational cost in such NA-MD, the program was modified. 
This modification uses an input parameter `params["do_rescaling"] ` ; that will change the number of SCF calculation.
If `params["do_rescaling"] =1` (as a default), the number is the same as that of TSH trajectories. Otherwise, the number is less than that of TSH trajectories.
Following this update, the code https://github.com/kosukesato/libra-code/blob/devel/src/libra_py/tsh.py was modified, too. I'll do pull-request about that modification if this pull-request is allowed to be merged.

# Minor change
`params["shift_E"]` was defined to shift excitation energies as we need. Note that the parameter influences only Libra-GAMESS part, and we don't have to set this parameter when we don't want to.

### CODE REVIEW

File: src/extract_gms.py [hunk 1]
```diff
@@ -362,15 +362,23 @@ def gms_extract(filename,states,min_shift,active_space,flag):
     # the total energies of excited states (1-electron basis)
 
     # *********Here, KS excitation energy -> total excitation energy***************
+    homo = active_space[0] - min_shift # HOMO : number ordering starts from 1, not 0.
```
Comment (alexvakimov) [inline src/extract_gms.py]: make sure this is the same convention that is used in Libra-QE as well, we want to keep the meaning of the "active_space" the same for all types of the interface

File: src/extract_gms.py [hunk 2]
```diff
@@ -362,15 +362,23 @@ def gms_extract(filename,states,min_shift,active_space,flag):
...
     nstates = len(states)
     E = MATRIX(nstates,nstates)
     for i in xrange(nstates):
-        h_indx = states[i].from_orbit[0] - min_shift  # index of the hole orbital w.r.t. the lowest included in the active space orbital
-        e_indx = states[i].to_orbit[0]   - min_shift  # --- same, only for the electron orbital
+        #h_indx = states[i].from_orbit[0] - min_shift  # index of the hole orbital w.r.t. the lowest included in the active space orbital
+        #e_indx = states[i].to_orbit[0]   - min_shift  # --- same, only for the electron orbital
+        #print "%i th excitation" %(i)
+        h_indx = states[i].from_orbit[0] + homo - 1
```
Comment (alexvakimov) [inline src/extract_gms.py]: these 2 lines + the line 365 - is it true? Why do we need the change with respect to the commented (now deleted) case?

File: src/hamiltonian_vib.py [hunk 3]
```diff
@@ -304,6 +304,14 @@ def update_vibronic_hamiltonian(ham_el, ham_vib, params,E_SD,nac,suffix, opt):
                     ham_vib.set(I,J,(-1.0j+0.0)*nac.get(I,J))
                     print "\n"
       
+
+    if "inactive_states" in params:
```
Comment (alexvakimov) [inline src/hamiltonian_vib.py]: that doesn't look like the sufficient check - one could just define such a variable, but this won't mean param["inactive_states"] are initialized. Better, use something like
if len(params["inactive_states"]) > 0

File: src/extract_gms.py (hunk 2 above)
Comment (kosukesato) [reply src/extract_gms.py]: This change of extract_gms.py is not needed. I'll take it back to previous version. Sorry for the inconvenience.

File: src/hamiltonian_vib.py (hunk 3 above)
Comment (kosukesato) [reply src/hamiltonian_vib.py]: Okay, I'll use if len(params["inactive_states"]) > 0 . Thank you.

**Pull request merged**